com.unity.render-pipelines.universal/ShaderLibrary/SSAO.hlsl

```diff
@@ -45,81 +91,77 @@ float4 _CameraViewZExtent[2];
 // Constants
 // kContrast determines the contrast of occlusion. This allows users to control over/under
 // occlusion. At the moment, this is not exposed to the editor because it's rarely useful.
-static const float kContrast = 0.6;
+static const half kContrast = half(0.6);
 
 // The constant below controls the geometry-awareness of the bilateral
 // filter. The higher value, the more sensitive it is.
-static const float kGeometryCoeff = 0.8;
+static const half kGeometryCoeff = half(0.8);
 
 // The constants below are used in the AO estimator. Beta is mainly used for suppressing
 // self-shadowing noise, and Epsilon is used to prevent calculation underflow. See the
-// paper (Morgan 2011 http://goo.gl/2iz3P) for further details of these constants.
-static const float kBeta = 0.002;
-#define EPSILON         1.0e-4
+// paper (Morgan 2011 https://bit.ly/3uAPRgz) for further details of these constants.
+static const half kBeta = half(0.002);
+static const half kEpsilon = half(0.0001);
 
 #if defined(USING_STEREO_MATRICES)
-#define unity_eyeIndex unity_StereoEyeIndex
+    #define unity_eyeIndex unity_StereoEyeIndex
 #else
-#define unity_eyeIndex 0
+    #define unity_eyeIndex 0
 #endif
 
-float4 PackAONormal(float ao, float3 n)
+half4 PackAONormal(half ao, half3 n)
 {
-    return float4(ao, n * 0.5 + 0.5);
+    return half4(ao, n * half(0.5) + half(0.5));
 }
 
-float3 GetPackedNormal(float4 p)
+half3 GetPackedNormal(half4 p)
 {
-    return p.gba * 2.0 - 1.0;
+    return p.gba * half(2.0) - half(1.0);
 }
 
-float GetPackedAO(float4 p)
+half GetPackedAO(half4 p)
 {
     return p.r;
 }
 
-float EncodeAO(float x)
+half EncodeAO(half x)
 {
     #if UNITY_COLORSPACE_GAMMA
-        return 1.0 - max(LinearToSRGB(1.0 - saturate(x)), 0.0);
+        return half(1.0 - max(LinearToSRGB(1.0 - saturate(x)), 0.0));
     #else
         return x;
     #endif
 }
 
-float CompareNormal(float3 d1, float3 d2)
+half CompareNormal(half3 d1, half3 d2)
 {
-    return smoothstep(kGeometryCoeff, 1.0, dot(d1, d2));
-}
-
-float2 GetScreenSpacePosition(float2 uv)
```

Comment:
Is this function being removed?

Reply:
Yes, I was doing that to clean up a little bit as it was only used in one place.
But to avoid any issues with users potentially using that function I'm adding it back in the next commit.